### DISCUSSION TITLE
Trying to get bottles to work in Debian Buster

### DISCUSSION THREAD

**Discussion by Lars** [bottlesdevs/Bottles#411] [Q&A]
Hello all.

I've been trying to get bottles to work on Debian Buster for a couple days now, almost a week.

All three varians (Snap, AppImage, .deb) fail for me. Snapd and AppImage both have some errors in regards to .so files so I guess it has to do with system dependencies. I think .deb would be the easiest to make work and stabler at that, but I have a different error, a python error that as far as I can tell has something to do with problems instantiating a gtk component in python, but as I've worked very little with Gtk in Python I'm not entirely sure:

`(bottles:8738): Gtk-CRITICAL **: 20:42:12.715: Error building template class 'BottlesList' for an instance of type 'BottlesList': .:69:1 Invalid object type 'HdyStatusPage'
Traceback (most recent call last):
  File "/usr/share/bottles/bottles/main.py", line 125, in do_command_line
    self.do_activate()
  File "/usr/share/bottles/bottles/main.py", line 169, in do_activate
    arg_lnk=self.arg_lnk)
  File "/usr/share/bottles/bottles/window.py", line 123, in __init__
    page_list = BottlesList(self, arg_executable)
  File "/usr/share/bottles/bottles/pages/list.py", line 207, in __init__
    self.btn_create.connect("pressed", self.window.show_add_view)
AttributeError: 'NoneType' object has no attribute 'connect'`

From what I see, that btn_create should be a Gtk.Template.Child() and it should in theory be working, but I've never used templates (Gtk in python) so I'm not sure if it's dependant on the .ui file meta/lamba-ed right before the class definition, and if it's a location issue, etc.

I appreciate any help to make this work. Bottles would really fit some use cases for me.

**Comment by mirkobrombin** [comment]
Hi, the problem is in the first row:
```
'BottlesList': .:69:1 Invalid object type 'HdyStatusPage' Traceback (most recent call last): File "/usr/share/bottles/bottles/main.py"
```
Bottles is missing the `libhandy-1-0` dependency which provide the `HdyWidgets`. There is a new build in the releases, can you give it a try? https://github.com/bottlesdevs/Bottles/releases/download/2021.7.28-treviso/com.usebottles.bottles_2021.7.28-treviso_amd64.deb

**Reply by Lars** [reply]
I am using that version, and I do have libhandy-1-0 installed (I even installed handy-1-demo to test it, and it works).

**Reply by mirkobrombin** [reply]
> I am using that version, and I do have libhandy-1-0 installed (I even installed handy-1-demo to test it, and it works).

try to install libhandy-dev

**Reply by Lars** [reply]
It's installed as well (already was) (libhandy-1-dev)

**Reply by mirkobrombin** [reply]
And the output doesn't change? It's weird :S

**Reply by mirkobrombin** [reply]
It say: `Invalid object type 'HdyStatusPage' Traceback` which mean "there is no support for HdyWidgets on your system"

**Reply by mirkobrombin** [reply]
Oh I got it! HdyStatusPage was introduced with libhandy 1.1, can check which version of the library is installed?

**Reply by Lars** [reply]
1.0, sadly.

I'm not sure I want to fight the Snap version, I'll first see how doable is to compile a libhandy just for bottles (if there's no backports).

Thanks.

**Reply by Lars** [reply]
No backports, either I compile one, or I'm toast, thanks for the help.

**Reply by mirkobrombin** [reply]
It is only provided by experimental
https://packages.debian.org/experimental/libhandy-1-dev

**Reply by Lars** [reply]
If I give up and try to fix a whole package instead, would you suggest wether appimage or the snap as the best option, please?

**Reply by mirkobrombin** [reply]
> If I give up and try to fix a whole package instead, would you suggest wether appimage or the snap as the best option, please?

AppImage if it works

**Reply by Lars** [reply]
Mirko, just to let you know, I built the package for 1.2.2 for Debian Buster (Stable), was easy, and so far bottles seems to be working, only, so you know, the glibc included in Buster is too old to run vagnilia 6.13, but it'll run Lutris 6.13 if you use it (I'll test Proton later).

**Reply by mirkobrombin** [reply]
> Mirko, just to let you know, I built the package for 1.2.2 for Debian Buster (Stable), was easy, and so far bottles seems to be working, only, so you know, the glibc included in Buster is too old to run vagnilia 6.13, but it'll run Lutris 6.13 if you use it (I'll test Proton later).

thanks
